### PR TITLE
Proximity sensing refactor

### LINKED ISSUE #26254: Conveyor belts are too fast for stacking machines
The stacking machine in maintenance can't keep up with the fastprocessing of conveyor belts, therefore its process won't grab the sheets created by the recycler.

### PR DESCRIPTION
**Pull request by Cyberboss**
@optimumtact 

Fixes #26254

### CODE REVIEW

File: code/game/objects/effects/proximity.dm [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		START_PROCESSING(SSobj, src)
+	SetRange(range)
+
+/datum/proximity_monitor/Destroy()
+	host = null
+	QDEL_LIST(checkers)
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/datum/proximity_monitor/process()
```
Comment (RemieRichards) [inline code/game/objects/effects/proximity.dm]: I said no polling
;_;

Comment (Cyberboss) [inline code/game/objects/effects/proximity.dm]: It was either poll or add checks to movable/Moved

Comment (Cyberboss) [inline code/game/objects/effects/proximity.dm]: This is for tracking when the sensor object moves

Comment (RemieRichards) [inline code/game/objects/effects/proximity.dm]: moved is reactive.
reactive > polling.

Comment (Cyberboss) [inline code/game/objects/effects/proximity.dm]: It's the option of polling for a few objects or be reactive for ALL objects

Comment (RemieRichards) [inline code/game/objects/effects/proximity.dm]: add hooks then

Comment (ChangelingRain) [inline code/game/objects/effects/proximity.dm]: Polling is very inaccurate. you're using SSobj, like, holy shit, that only runs every two seconds. You can move 20 tiles in that time!

File: code/game/machinery/camera/camera.dm [hunk 2]
```diff
@@ -59,13 +60,9 @@
...
 		toggle_cam()
 
-/obj/machinery/camera/Move()
-	remove_from_proximity_list(src, 1)
-	return ..()
-
 /obj/machinery/camera/Destroy()
 	toggle_cam(null, 0) //kick anyone viewing out
-	remove_from_proximity_list(src, 1)
+	QDEL_NULL(proximity_monitor)
```
Comment (RemieRichards) [inline code/game/machinery/camera/camera.dm]: this should probably be on /atom/movable/Destroy, don't even give people the opportunity to forget to gc it.

**Comment by coiax** [comment]
Make the abstract objects indestructible and acid resistant and all that. In addition, make the proximity abstract effects have the proximity sensor item sprite and set their invisibility to abstract, so you can make them visible for debugging purposes.

**Comment by coiax** [comment]
Actually, I'm not sure why abstract needs to be a child of effect.

**Comment by RemieRichards** [comment]
@coiax a lot of effects are already indestructible in certain ways, so inheriting from them just inherits those.

**Pull request merged**